crates/common/mqtt_channel/src/tests.rs

```diff
@@ -2,17 +2,16 @@
 mod tests {
     use crate::*;
     use futures::{SinkExt, StreamExt};
-    use serial_test::serial;
+    use mqtt_tests::test_mqtt_server::MqttProcessHandler;
     use std::convert::TryInto;
     use std::time::Duration;
 
     const TIMEOUT: Duration = Duration::from_millis(1000);
 
     #[tokio::test]
-    #[serial]
     async fn subscribing_to_messages() -> Result<(), anyhow::Error> {
         // Given an MQTT broker
-        let broker = mqtt_tests::test_mqtt_broker();
+        let broker = MqttProcessHandler::new(55551, 3551);
```

Comment:
With the proposed approach, each test would have to come with genuinely new port numbers. In order to remove this burden can we instead have the `mqtt_tests::test_mqtt_broker()` function generating those, say using an internal sequence?

Reply:
The problem with that is that we cannot have all tests in one binary. That means if cargo runs multiple binaries in parallel (which as I believe it does), we would still run into conflicts.

Reply:
Agree with @didier-wenzek. Expecting the tests to come up with unique ports doesn't seem practical, especially if tests in different crates are going to run in parallel. Then the developer adding a new test would have to scan through the ports used by all other tests in all other modules and then come up with a new one. That doesn't sound practical.

Apart from that, I'm assuming that starting the broker per test would increase the running time of individual tests, but it will be compensated in the overall test running time as these tests can now run in parallel.